src/model/transaction/moveBlockInContentState.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule moveBlockInContentState
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+const invariant = require('invariant');
+
+import type ContentBlock from 'ContentBlock';
+import type ContentState from 'ContentState';
+import type {DraftInsertionType} from 'DraftInsertionType';
+
+function moveBlockInContentState(
+  contentState: ContentState,
+  blockToBeMoved: ContentBlock,
+  targetBlock: ContentBlock,
+  insertionMode: DraftInsertionType
+): ContentState {
+  invariant(
+    blockToBeMoved.getKey() !== targetBlock.getKey(),
+    'Block cannot be moved next to itself.'
+  );
+
+  invariant(
+    insertionMode !== 'replace',
+    'Replacing blocks is not supported.'
+  );
```

Comment:
replace is needed in AtomicBlockUtils but not here. Actually i could just implement replace here as well...